### ISSUE TITLE
Tests are failing on GCC ≥ 12 in `test_attr_color`

### ISSUE THREAD

**Issue by RaitoBezarius** [neomutt/neomutt#4128]
## Expected Behaviour

Passing test suite.

## Actual Behaviour

https://0x0.st/H3MX.txt without UBSAN **or** ASAN or **debug build**.
**Passing** with UBSAN **or** ASAN or **with debug**.

## Steps to Reproduce

Unfortunately, NixOS environment reproduces this fairly easily, I have yet to try another Linux distribution than NixOS with a fairly recent compiler and try to reproduce this.

## How often does this happen?

- Always

## When did it start to happen?

- When I upgraded

It appeared in the last release version.

## NeoMutt Version

```
NeoMutt 20231103
Copyright (C) 1996-2022 Michael R. Elkins and others.
NeoMutt comes with ABSOLUTELY NO WARRANTY; for details type 'neomutt -vv'.
NeoMutt is free software, and you are welcome to redistribute it
under certain conditions; type 'neomutt -vv' for details.

System: Linux 6.5.12 (x86_64)
ncurses: ncurses 6.4.20221231 (compiled with 6.4.20221231)
libidn2: 2.3.4 (compiled with 2.3.4)
GPGME: 1.23.0-unknown
OpenSSL: OpenSSL 3.0.12 24 Oct 2023
libnotmuch: 5.6.0
storage: lmdb
compression: zlib, zstd

Configure options: --prefix=/nix/store/qprbzgjk6l4wpnk7i1s7jgi853731cxz-neomutt-20231103 --enable-autocrypt --gpgme --gss --lmdb --notmuch --ssl --sasl --with-homespool=mailbox --with-mailpath= --disable-include-path-in-cflags --zlib --zstd

Compilation CFLAGS: -fno-delete-null-pointer-checks -D_ALL_SOURCE=1 -D_GNU_SOURCE=1 -D__EXTENSIONS__ -D_XOPEN_SOURCE_EXTENDED -DNCURSES_WIDECHAR -O2

Default options:
  +attach_headers_color +compose_to_sender +compress +cond_date +debug 
  +encrypt_to_self +forgotten_attachments +forwref +ifdef +imap +index_color 
  +initials +limit_current_thread +multiple_fcc +nested_if +new_mail +nntp +pop 
  +progress +quasi_delete +regcomp +reply_with_xorig +sensible_browser +sidebar 
  +skip_quoted +smtp +status_color +timeout +tls_sni +trash 

Compile options:
  +autocrypt +fcntl -flock -fmemopen +futimens +getaddrinfo -gnutls +gpgme 
  -gsasl +gss +hcache +homespool +idn +inotify -locales_hack -lua -mixmaster 
  +nls +notmuch +openssl +pgp +regex +sasl +smime +sqlite +sun_attachment 
  +truecolor 

MAILPATH="mailbox"
PKGDATADIR="/nix/store/qprbzgjk6l4wpnk7i1s7jgi853731cxz-neomutt-20231103/share/neomutt"
SENDMAIL="sendmail"
SYSCONFDIR="/nix/store/qprbzgjk6l4wpnk7i1s7jgi853731cxz-neomutt-20231103/etc"

To learn more about NeoMutt, visit: https://neomutt.org
If you find a bug in NeoMutt, please raise an issue at:
    https://github.com/neomutt/neomutt/issues
or send an email to: <neomutt-devel@neomutt.org>
```

## Extra Info

* Operating System and its version: NixOS unstable.
* Were you using multiple copies of NeoMutt at once? No.
* Were you using 'screen' or 'tmux'? No.
* Is your email local (maildir) or remote (IMAP)? Irrelevant.

Related to https://github.com/NixOS/nixpkgs/pull/262271.

**Comment by flatcap** [comment]
Yes, sorry about that.
You can either ignore that failure, or cherry-pick the fix, 24f8644c2, that was merged just after the release.

**Comment by RaitoBezarius** [comment]
Thank you, I can close, this is addressed now.
